### PR TITLE
Modernize and remove Python 2.x compatibility code

### PR DESCRIPTION
**Pull request by intgr**
If you feel like it, let's get this one merged first before #339.

### CODE REVIEW

File: tests/test_dummy.py [hunk 1]
```diff
@@ -1,6 +0,0 @@
-from django.test import TestCase
-
-
-class TestDummy(TestCase):
-    def test_one_plus_one(self):
-        assert 1 + 1 == 2
```
Comment (intgr) [inline tests/test_dummy.py:6]: This test seemed rather useless 😛 but I can restore it if there's any reason.

**Comment by intgr** [comment]
Ping @alanjds, any thoughts on this?

File: rest_framework_nested/relations.py [hunk 2]
```diff
@@ -26,7 +19,7 @@ class NestedHyperlinkedRelatedField(rest_framework.relations.HyperlinkedRelatedF
 
     def __init__(self, *args, **kwargs):
         self.parent_lookup_kwargs = kwargs.pop('parent_lookup_kwargs', self.parent_lookup_kwargs)
-        super(NestedHyperlinkedRelatedField, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
```
Comment (alanjds) [inline rest_framework_nested/relations.py:22]: That part can stay. It helps typing tools.
```suggestion
        super(NestedHyperlinkedRelatedField, self).__init__(*args, **kwargs)
```

File: rest_framework_nested/relations.py [hunk 3]
```diff
@@ -103,4 +96,4 @@ def __init__(self, view_name=None, **kwargs):
         assert view_name is not None, 'The `view_name` argument is required.'
         kwargs['read_only'] = True
         kwargs['source'] = '*'
-        super(NestedHyperlinkedIdentityField, self).__init__(view_name=view_name, **kwargs)
+        super().__init__(view_name=view_name, **kwargs)
```
Comment (alanjds) [inline rest_framework_nested/relations.py:99]: That part can stay. It helps typing tools.
```suggestion
        super(NestedHyperlinkedIdentityField, self).__init__(view_name=view_name, **kwargs)
```

File: rest_framework_nested/serializers.py [hunk 4]
```diff
@@ -27,10 +27,10 @@ class NestedHyperlinkedModelSerializer(rest_framework.serializers.HyperlinkedMod
 
     def __init__(self, *args, **kwargs):
         self.parent_lookup_kwargs = kwargs.pop('parent_lookup_kwargs', self.parent_lookup_kwargs)
-        super(NestedHyperlinkedModelSerializer, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
```
Comment (alanjds) [inline rest_framework_nested/serializers.py:30]: That part can stay. It helps typing tools.
```suggestion
        super(NestedHyperlinkedModelSerializer, self).__init__(*args, **kwargs)
```

File: rest_framework_nested/serializers.py [hunk 5]
```diff
@@ -27,10 +27,10 @@ class NestedHyperlinkedModelSerializer(rest_framework.serializers.HyperlinkedMod
...
 
     def build_url_field(self, field_name, model_class):
-        field_class, field_kwargs = super(NestedHyperlinkedModelSerializer, self).build_url_field(
+        field_class, field_kwargs = super().build_url_field(
```
Comment (alanjds) [inline rest_framework_nested/serializers.py:33]: That part can stay. It helps typing tools.
```suggestion
        field_class, field_kwargs = super(NestedHyperlinkedModelSerializer, self).build_url_field(
```

File: rest_framework_nested/viewsets.py [hunk 6]
```diff
@@ -44,7 +44,7 @@ def get_queryset(self):
         Filter the `QuerySet` based on its parents as defined in the
         `serializer_class.parent_lookup_kwargs` or `viewset.parent_lookup_kwargs`
         """
-        queryset = super(NestedViewSetMixin, self).get_queryset()
+        queryset = super().get_queryset()
```
Comment (alanjds) [inline rest_framework_nested/viewsets.py:47]: That part can stay. It helps typing tools.
```suggestion
        queryset = super(NestedViewSetMixin, self).get_queryset()
```

File: tests/models.py [hunk 7]
```diff
@@ -11,7 +10,7 @@ class CustomField(models.CharField):
 
     def __init__(self, *args, **kwargs):
         kwargs['max_length'] = 12
-        super(CustomField, self).__init__(*args, **kwargs)
+        super().__init__(*args, **kwargs)
```
Comment (alanjds) [inline tests/models.py:13]: That part can stay. It helps typing tools.
```suggestion
        super(CustomField, self).__init__(*args, **kwargs)
```

**Review by alanjds — APPROVED** [review]
Thanks for removing Python 2 compat code.

I left only some stuff that still works with Py 3 and helps on typing for some IDEs

File: rest_framework_nested/relations.py (hunk 2 above)
Comment (intgr) [reply rest_framework_nested/relations.py:22]: I don't mind keeping these, but I doubt it's necessary really.

Which typing tools? I would think they are capable of finding the base class from the `class ...(...):` definition.

Comment (alanjds) [reply rest_framework_nested/relations.py:22]: Looks like _at least_ VSCode have trouble detecting upper classes in some cases, if the middle classes do not put its names in the `super()`.

Well, lets have this merged and revert back if some complains arise :+1:

File: tests/test_dummy.py (hunk 1 above)
Comment (alanjds) [reply tests/test_dummy.py:6]: Yeah, this one I put on new projects to be sure that the test infra is collecting the tests correctly. Can be removed.

**Pull request merged**